### PR TITLE
Closes #5268: Filter out private tabs in RemoteTabsFeature

### LINKED ISSUE #5268: Filter-out private tabs in remote tabs

### PR DESCRIPTION
**Pull request by eoger**
r? @jonalmeida

### CODE REVIEW

File: components/feature/remotetabs/src/main/java/mozilla/components/feature/remotetabs/RemoteTabsFeature.kt [hunk 1]
```diff
@@ -40,7 +40,10 @@ class RemoteTabsFeature(
                 val lastUsed = 0L
                 // TO-DO: https://github.com/mozilla-mobile/android-components/issues/5179
                 val iconUrl = null
-                state.tabs.map { tab ->
+                state.tabs.mapNotNull { tab ->
```
Comment (jonalmeida) [inline components/feature/remotetabs/src/main/java/mozilla/components/feature/remotetabs/RemoteTabsFeature.kt]: Couldn't we just stick a filter before it or did I misunderstand what we wanted to do? 🤔 

```kotlin
state.tabs.filter { !it.content.private }
```

Comment (eoger) [reply components/feature/remotetabs/src/main/java/mozilla/components/feature/remotetabs/RemoteTabsFeature.kt]: Yup that works too, I was wondering if the compiler would optimize the two array iterations.

**Comment by jonalmeida** [comment]
SSL error when downloading gradle for the ui-icons build:
```
[task 2019-12-10T16:24:48.150Z] Exception in thread "main" javax.net.ssl.SSLHandshakeException: Remote host closed connection during handshake
```

File: components/feature/remotetabs/src/main/java/mozilla/components/feature/remotetabs/RemoteTabsFeature.kt (hunk 1 above)
Comment (jonalmeida) [reply components/feature/remotetabs/src/main/java/mozilla/components/feature/remotetabs/RemoteTabsFeature.kt]: The Kotlin compiler is quite good at generating optimized bytecode, and the AOT compiler on devices do even more things to reduce runtime costs, so in general, I don't think we need to think about those kinds of optimizations on Android any more.

**Pull request merged**